plugins/amazonq/shared/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/project/ProjectContextEditorListener.kt

```diff
@@ -2,15 +2,23 @@
 // SPDX-License-Identifier: Apache-2.0
 
 package software.aws.toolkits.jetbrains.services.amazonq.project
-import com.intellij.openapi.fileEditor.FileEditorManager
+
+import com.intellij.openapi.fileEditor.FileDocumentManager
+import com.intellij.openapi.fileEditor.FileEditorManagerEvent
 import com.intellij.openapi.fileEditor.FileEditorManagerListener
-import com.intellij.openapi.vfs.VirtualFile
-import software.aws.toolkits.jetbrains.settings.CodeWhispererSettings
 
 class ProjectContextEditorListener : FileEditorManagerListener {
-    override fun fileClosed(source: FileEditorManager, file: VirtualFile) {
-        if (CodeWhispererSettings.getInstance().isProjectContextEnabled()) {
-            ProjectContextController.getInstance(source.project).updateIndex(file.path)
+    override fun selectionChanged(event: FileEditorManagerEvent) {
```

Comment:
When user is editing current file, don't call the updateIndex API. This will create too much load for the LSP to handle. Call update Index API only when user finishes editing current file

Reply:
currently update will be called when the active editor is changed (thus selectionChanged)